src/Native/System.Native/pal_networking.cpp

```diff
@@ -1215,9 +1223,14 @@ extern "C" Error SystemNative_SetIPv4MulticastOption(int32_t socket, int32_t mul
         return PAL_EINVAL;
     }
 
+#if HAVE_IP_MREQN
     ip_mreqn opt = {.imr_multiaddr = {.s_addr = option->MulticastAddress},
                     .imr_address = {.s_addr = option->LocalAddress},
                     .imr_ifindex = option->InterfaceIndex};
+#else
+    ip_mreq opt = {.imr_multiaddr = {.s_addr = option->MulticastAddress},
+                   .imr_interface = {.s_addr = option->LocalAddress}};
+#endif
```

Comment:
Yeah, that's what I was getting at with my question on the other case of this.  Or, if there's some default value we can set it to, do so, and then the managed code can check for that default and do whatever's reasonable, including potentially erroring out.